apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c

```diff
@@ -634,6 +634,10 @@ static void utc_audio_pcm_readi_p(void)
 		remain -= frames_read;
 		ret = write(fd, buffer, bytes_per_frame * frames_read);
 		TC_ASSERT_EQ_CLEANUP("pcm_readi", ret, (bytes_per_frame * frames_read), clean_all_data(fd, buffer));
+		if (remain < pcm_get_buffer_size(g_pcm)) {
+			pcm_drop(g_pcm);
+			break;
+		}
```

Comment:
Regarding the pcm_drop in readi utc.. 
What you have done is ok... it can also be done like one of these..
1. just break from the loop if remain < pcm_get_buffer_size(g_pcm) .. pcm_drop is not required... 
2. Rearrange the code like below:

if (remain < frames_read)
    ret = write(fd, buffer, bytes_per_frame * remain);
else
    ret = write(fd, buffer, bytes_per_frame * frames_read};
    
remain -= frames_read;

I think method 1 is better.